### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by dokshina**

### CODE REVIEW

File: test/python/project.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+    # add custom-project to version_file_keys
+    project.version_file_keys.add(SUBMODULE_NAME)
+
+
+ignored_data = [
+    {'dir': '',                  'file': 'ignored.txt'},
+    {'dir': '',                  'file': 'asterisk'},
+    {'dir': '',                  'file': 'ignored.lua'},
+    {'dir': '',                  'file':  'ignored_by.format'},
```
Comment (olegrok) [inline test/python/project.py]: Redundant space between `:` - `'file':  'ignored_by.format'}`

File: test/python/test_docker_pack.py [hunk 2]
```diff
@@ -244,7 +237,7 @@ def test_docker_e2e(project, docker_image, tmpdir, docker_client):
         image_name,
         environment=environment,
         ports={'8082': '8082'},
-        name='{}-instance-1'.format(project['name']),
```
Comment (olegrok) [inline test/python/test_docker_pack.py]: Why do you remove a `project` from this test?

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -42,6 +42,8 @@ variables:
 prebuild:
   extends: .dind
   stage: prebuild
+  variables:
+    DOCKER_BUILDKIT: 1
```
Comment (olegrok) [inline .gitlab-ci.yml]: May be define it in `.dind`?

File: test/python/conftest.py [hunk 4]
```diff
@@ -1,256 +1,133 @@
...
+# ################
+# Project fixtures
+# ################
+# There are three types of projects:
+# * light_project ({original,deprecated}_light_project):
+#   Cartridge CLI creates project with cartridge dependency by default.
+#   It's known that installing cartridge rocks is a long operation,
+#   so we don't want to perform it on every test.
+#   These fixtures are used to decrease packing time.
+#   They don't have cartridge dependency,
```
Comment (olegrok) [inline test/python/conftest.py]: nit: "don't have `a/the` cartridge"

File: test/python/project.py [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+
+from utils import create_project
+from utils import recursive_listdir
+from utils import tarantool_enterprise_is_used
+
+
+CARTRIDGE_PACK_SPECIAL_FILES = {
+    'cartridge.pre-build',
+    'cartridge.post-build',
+    '.cartridge.ignore',
```
Comment (olegrok) [inline test/python/project.py]: May be it's better to add a note that two last files are from "deprecated" mode

File: test/python/test_docker_pack.py [hunk 6]
```diff
@@ -150,9 +146,9 @@ def test_docker_pack(project, docker_image, tmpdir, docker_client):
     os.remove(arhive_path)
 
     assert_dir_contents(
-        files_list=recursive_listdir(os.path.join(tmpdir, 'usr/share/tarantool/', project['name'])),
-        exp_files_list=project['distribution_files_list'],
-        exp_rocks_content=project['rocks_content'],
+        files_list=recursive_listdir(os.path.join(tmpdir, 'usr/share/tarantool/', project.name)),
+        exp_files_list=project.distribution_files,
```
Comment (olegrok) [inline test/python/test_docker_pack.py]: usually `exp` is shortcut for `exponent`. Let's call it explicitly `expected`

File: test/python/test_docker_pack.py (hunk 2 above)
Comment (dokshina) [reply test/python/test_docker_pack.py]: To simplify the test.

Comment (dokshina) [reply test/python/test_docker_pack.py]: The project is created on `docker_image` fixture setup.

Comment (olegrok) [reply test/python/test_docker_pack.py]: May be pass `project name` with image name to eliminate such hardcode?

Comment (dokshina) [reply test/python/test_docker_pack.py]: Yes, I think that docker_image object can contain field `project_name`.

File: test/python/project.py [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+        # create project and save its path
+        self.path = create_project(basepath, name, template)
+
+        # project files list
+        self.project_files = recursive_listdir(self.path)
+
+        # files that should be delivered in the result package
+        self.distribution_files = filter_out_files_removed_on_pack(self.project_files)
+        self.distribution_files.add('VERSION')
+        if tarantool_enterprise_is_used():
```
Comment (olegrok) [inline test/python/project.py]: You use this function twice in this ... function

May be let's store it to var and reuse

File: test/python/project.py [hunk 8]
```diff
@@ -59,10 +59,13 @@ def __init__(self, name, basepath, template='cartridge'):
         # project files list
         self.project_files = recursive_listdir(self.path)
 
+        # save tarantool_enterprise_is_used() result to variable
```
Comment (olegrok) [inline test/python/project.py]: It's a bit trivial

**Pull request merged**